spec/requests/admin/plugins_controller_spec.rb

```diff
@@ -77,6 +77,22 @@
         expect(response.status).to eq(404)
         expect(response.parsed_body["errors"]).to include(I18n.t("not_found"))
       end
+
+      it "404s if the plugin is not visible" do
+        poll = Discourse.plugins_by_name["poll"]
+        poll.stubs(:visible?).returns(false)
+
+        get "/admin/plugins/poll.json"
+        expect(response.status).to eq(404)
+      end
+
+      it "404s if the plugin is not configurable" do
+        poll = Discourse.plugins_by_name["poll"]
+        poll.stubs(:configurable?).returns(false)
+
+        get "/admin/plugins/poll.json"
+        expect(response.status).to eq(404)
+      end
```

Comment:
```suggestion
```

I don't think we need this since `configurable?` is an internal concern of the `visible?` method.